theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -687,6 +687,24 @@ def test_bilinear_upsampling_1D(self):
             up_mat_2d = self.get_upsampled_twobytwo_mat(input_x, ratio)
             utt.assert_allclose(f(), up_mat_2d, rtol=1e-06)
 
+    def test_bilinear_upsampling_reshaping(self):
+        """Test bilinear upsampling without giving shape information
+
+        This method tests the bilinear_upsampling method
+        without giving batch_size and num_input_channels
+
+        """
+        # upsampling for a ratio of two
+        input_x = np.array([[[[1, 2], [3, 4]]]], dtype=theano.config.floatX)
+
+        for ratio in [2, 3]:
+            for use_1D_kernel in [True, False]:
+                bilin_mat = bilinear_upsampling(input=input_x, ratio=ratio,
```

Comment:
Please set both `batch_size` and `num_input_channels` to `None`, so that in case the default values are changed in the future the test remains valid.